### ISSUE TITLE
Popup menu node not taking relative position 

### ISSUE THREAD

**Issue by deermen001** [godotengine/godot#38171]
___

***Note:** This issue has been confirmed several times already. No need to confirm it further.*

___

OS : Windows 10
Godot version : 3.2
[Issue.zip](https://github.com/godotengine/godot/files/4528839/Issue.zip)

I attached popup menu as child node of a 2d node. I moved the parent node with which the popup menu node moved in the editor but on running it still shows it at previous position.

I think the popup menu relative position property is interpreted as absolute global position.

**Comment by Calinou** [comment]
@deermen001 Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

Also, which Godot version and operating system are you using? (Please fill in the issue template for future issues.)

**Comment by deermen001** [comment]
> @deermen001 Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.
> 
> Also, which Godot version and operating system are you using? (Please fill in the issue template if you plan on reporting other issues.)

I edited my comment as per your guidelines. Please correct me if there is still something wrong.

**Comment by alexontheweb** [comment]
I second the issue. Both PopupMenu and PopupPanel don't use relative to parent position.

**Godot version:** Godot Engine v3.2.3.stable.official
**OS/device including version:** Windows 7

**Steps to reproduce:**
1. Create a scene:
```
- Node2D 
-- PopupMenu (or PopupPanel)
```
2. Move Node2D away from default top left corner. Editor shows child popup node moves along with parent as expected.
3. Run project: Popup node would appear at window's top left corner instead of near parent node.

**Minimal reproduction project:** [popup_issue.zip](https://github.com/godotengine/godot/files/5740504/popup_issue.zip)

**Comment by abardess** [comment]
As of 3/3/2021 this is still an issue. The PopupMenu node does not seem to inherent relative position from its parent.

**Godot Version:** Godot Engine v3.2.3.stable.official
**OS:** Windows 10

I am able to work around it by making a function to set the inheritance manually in its parent:

Scene:
` -Node2D
--Popup Menu`

Script attached to Node2D:
` extends Node

onready var popup = $PopupMenu

func _process(delta):
     popup.rect_global_position = self.position
`

Alternatively, if you don't need to update the position every frame, you can put "popup.rect_global_position = self.position" right before you need to show the actual popup, assuming that behavior is controlled from the parent node.

**Comment by JonathanGrant92** [comment]
This is an annoying problem. I want the popup to popup, _aside: popup() for vector2 parameter should be an issue_, above the button, not over it. So, I worked out a solution that is dynamic to changing game window resolutions/resizing.

`var menu_position
var vertical_offset = 0
onready var zoning_button = get_node("Zoning")
onready var zoning_menu = get_node("Zoning/ZoningMenu")

func _ready():
	menu_position = zoning_button.rect_position
	vertical_offset = zoning_button.rect_size.y + zoning_menu.rect_size.y

func _on_Zoning_pressed():
	menu_position.y = OS.get_window_safe_area().size.y - vertical_offset
	zoning_menu.set_position(menu_position)
	zoning_menu.show()`

Feels like a hacky workaround, but it works for Godot 3.3.2!

OS.get_window_safe_area() is a very help function. which helped me solve many positioning problems in past projects. Probably should not be the go to function, but a fix to position vexing nodes like Popup and its derivatives. Best to use Layout button and container nodes for positioning gui.

**Comment by Alec-15** [comment]
I'm having the same issue in 3.3.3 stable on macos

**Comment by BirkTorpmannHagen** [comment]
This still does not seem to be solved.

**Comment by kongehund** [comment]
I also got this bug.

This may or may not be helpful for the devs when debugging:

I only started encountering this bug once I created a Window `HelpWindow` somewhere else in the scene tree. Before that, the Position of i.e. `File` was relative to `FileButton`.

![image](https://github.com/godotengine/godot/assets/63306696/fb881713-b883-4984-8547-19b633fc5ff3)

**Comment by rakete** [comment]
I ran into the same problem, I found this old PR: #78269 which fixes the problem for MenuButton, but not for MenuBar. I ported the fix over to menu_bar.cpp and it seems to work fine.

However, the fix still has issues, it works correctly afterwards when I am zoom the viewport and have the MenuBar/MenuButton on a Canvas. But if I don't have them on a Canvas, and so I want them to be affected by the zoom, then the popup isn't scaled anymore.

I assume this line:
```
Size2 size = get_size() * get_viewport()->get_canvas_transform().get_scale();
```
should be kept and somehow should only be applied when we're not on a canvas. Or maybe `get_canvas_transform().get_scale()` should not be affected by the zoom of the camera that is _not_ on this canvas?

**Comment by KoBeWi** [comment]
PopupMenu inherits Window now, so it's expected that it won't be local to Control nodes. The position is also consistent between editor and runtime.

> I only started encountering this bug once I created a Window HelpWindow somewhere else in the scene tree. Before that, the Position of i.e. File was relative to FileButton.

How? It shouldn't be possible.

**Comment by kongehund** [comment]
> How? It shouldn't be possible.
In that case, most likely because the FileButton is in the top left of my window (likely with no margins, as I recall), meaning the coordinates would look the same regardless if they're local to main Window or FileButton

**Comment by KoBeWi** [comment]
Yeah, the FileButton is in the top-left corner, so saying that File has relative position if its parent does not move is not really correct.
Closing per my previous comment.
